numpy/_core/src/umath/loops_trigonometric.dispatch.cpp

```diff
@@ -0,0 +1,269 @@
+#include "simd/simd.h"
+#include "loops_utils.h"
+#include "loops.h"
+#include "fast_loop_macros.h"
+#include <hwy/highway.h>
+namespace hn = hwy::HWY_NAMESPACE;
+
+/*
+ * Vectorized approximate sine/cosine algorithms: The following code is a
+ * vectorized version of the algorithm presented here:
+ * https://stackoverflow.com/questions/30463616/payne-hanek-algorithm-implementation-in-c/30465751#30465751
+ * (1) Load data in registers and generate mask for elements that are within
+ * range [-71476.0625f, 71476.0625f] for cosine and [-117435.992f, 117435.992f]
+ * for sine.
+ * (2) For elements within range, perform range reduction using
+ * Cody-Waite's method: x* = x - y*PI/2, where y = rint(x*2/PI). x* \in [-PI/4,
+ * PI/4].
+ * (3) Map cos(x) to (+/-)sine or (+/-)cosine of x* based on the
+ * quadrant k = int(y).
+ * (4) For elements outside that range, Cody-Waite
+ * reduction performs poorly leading to catastrophic cancellation. We compute
+ * cosine by calling glibc in a scalar fashion.
+ * (5) Vectorized implementation
+ * has a max ULP of 1.49 and performs at least 5-7x(x86) - 2.5-3x(Power) -
+ * 1-2x(Arm) faster than scalar implementations when magnitude of all elements
+ * in the array < 71476.0625f (117435.992f for sine).  Worst case performance
+ * is when all the elements are large leading to about 1-2% reduction in
+ * performance.
+ * TODO: use vectorized version of Payne-Hanek style reduction for large
+ * elements or when there's no native FUSED support instead of fallback to libc
+ */
+
+#if HWY_NATIVE_FMA  // native support
```

Comment:
Looks like we can't use this just yet, as if you build NumPy with no CPU features on AArch64 then Highway detects ASIMD anyway - so I think we need to revert this back to using NumPy's gating until we implement the new `HWY_` targets?

Reply:
This is why we're seeing undefined `!npyv_loadable_stride_f32`, as there's no `NPY_HAVE_ASIMD` in the baseline.

Reply:
Do you want me to revert it to `#ifdef NPY_SIMD_FMA3`?

Reply:
> Do you want me to revert it to `#ifdef NPY_SIMD_FMA3`?

Maybe we can try something like `#if defined(NPY_SIMD) && HWY_NATIVE_FMA` ? That'd check if NumPy thinks it is using SIMD and then let HWY decide if it's using FMA 🤔

Reply:
If we are using `NPY_SIMD`, might as well use `NPY_SIMD_FMA3`. Lets give it a try.

Reply:
@r-devulap, they're not quite the same, `NPY_SIMD` means NumPy expects SIMD, Highway and NumPy might disagree on FMA support. I don't mind though, NumPy's FMA3 should probably override Highways FMA at the moment anyway.